sql/src/main/java/io/crate/expression/operator/CIDROperator.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.expression.operator;
+
+import org.elasticsearch.common.collect.Tuple;
+import org.elasticsearch.common.network.InetAddresses;
+
+import java.math.BigInteger;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.nio.ByteBuffer;
+
+public final class CIDROperator {
+
+    public static final String LLT = Operator.PREFIX + "<<";
+
+    private static final int IPV4_ADDRESS_LEN = 4;
+    private static final int IPV6_ADDRESS_LEN = 16;
+
+    public static void register(OperatorModule module) {
+        module.registerDynamicOperatorFunction(LLT, CmpOperatorResolver.forIpCmpOperator(LLT, CIDROperator::llt));
+    }
+
+    public static boolean llt(String ipStr, String cidr) {
+        if (null == ipStr || null == cidr) {
+            throw new IllegalArgumentException();
+        }
+        boolean leftHasSlash = ipStr.contains("/");
+        boolean rightHasSlash = cidr.contains("/");
+        if (leftHasSlash == rightHasSlash) {
+            return false;
+        }
+        if (false == rightHasSlash || leftHasSlash) {
+            throw new IllegalArgumentException("operands are in the wrong order");
+        }
+        try {
+            BigInteger ip = toBigIntegerIp(InetAddress.getByName(ipStr));
+            return extractRange(cidr, (start, end) -> start.compareTo(ip) <= 0 && ip.compareTo(end) < 0);
+        } catch (UnknownHostException uhe) {
+            throw new IllegalArgumentException(uhe);
+        }
+    }
+
+    public static Tuple<InetAddress, InetAddress> extractRange(String cidr) {
+        return extractRange(cidr, (start, end) -> new Tuple<>(fromBigIntegerIp(start), fromBigIntegerIp(end)));
+    }
+
+    @FunctionalInterface
+    private interface IpRange<R> {
```

Comment:
`IpRange` seems a bit weird to me as a name for a `FunctionalInterface`. How does one "apply" a range?